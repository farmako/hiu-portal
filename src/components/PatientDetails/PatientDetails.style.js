import styled from 'styled-components';

const PatientDetailsStyles = styled.div`
  .patient-details-container {
    font-family: "Roboto", sans-serif;
    background: #f1f1f1;
    display: flex;
    height: 32px;
    padding: 0 10px;
    margin-top: 5px;
    align-items: center;
  }
`;

export default PatientDetailsStyles;
