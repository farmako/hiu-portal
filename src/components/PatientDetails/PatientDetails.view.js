import React from 'react';
import * as PropTypes from 'prop-types';
import PatientDetailsStyles from './PatientDetails.style';

const PatientDetails = ({ firstName }) => (
  <PatientDetailsStyles>
    <div className="patient-details-container">
      <span>
        Name :
        {firstName}
      </span>
    </div>
  </PatientDetailsStyles>
);

PatientDetails.propTypes = {
  firstName: PropTypes.string,
};

PatientDetails.defaultProps = {
  firstName: '',
};

export default PatientDetails;
