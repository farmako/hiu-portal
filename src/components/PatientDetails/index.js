import PatientDetails from './PatientDetails.view';

export default PatientDetails;
