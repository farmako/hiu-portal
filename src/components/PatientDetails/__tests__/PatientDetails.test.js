import React from 'react';
import {shallow} from "enzyme";
import PatientDetails from "../PatientDetails.view";

describe('PatientDetails', () => {
    const wrapper = shallow(
      <PatientDetails firstName="John" />
    );

    it('should render properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render name with firstname', () => {
        expect(wrapper.find("span").at(0).text()).toEqual('Name :John')
    });
});