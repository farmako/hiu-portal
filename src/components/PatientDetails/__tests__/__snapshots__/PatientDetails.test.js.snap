// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientDetails should render properly 1`] = `
<styled.div>
  <div
    className="patient-details-container"
  >
    <span>
      Name :
      John
    </span>
  </div>
</styled.div>
`;
